### PR TITLE
Creating API endpoints

### PR DESCRIPTION
**Pull request by bustyAI**
1. Updating JPA entity relationships
2. StudentController completed
3. AssignmentController completed
4. EnrollmentController completed

### CODE REVIEW

**Comment by Br-ak** [comment]
Went through and tested StudentController and EnrollmentController with Postman and found no issues.

File: src/main/java/com/cst438/controller/StudentController.java [hunk 1]
```diff
@@ -16,66 +20,182 @@
...
+
+    // student gets transcript showing list of all enrollments
+    // studentId will be temporary until Login security is implemented
+    // example URL /transcript?studentId=19803
+    @GetMapping("/transcript")
+    public List<EnrollmentDTO> getTranscript(@RequestParam("studentId") int studentId) {
+        // TODO
+        // list course_id, sec_id, title, credit, grade in chronological order
+        // user must be a student
+        // hint: use enrollment repository method findEnrollmentByStudentIdOrderByTermId
```
Comment (bustyAI) [inline src/main/java/com/cst438/controller/StudentController.java:43]: Could delete these comments

File: src/main/java/com/cst438/controller/StudentController.java [hunk 2]
```diff
@@ -16,66 +20,182 @@
...
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND,
+                    "No enrollments found for student Id: " + studentId);
+        }
+        List<EnrollmentDTO> elist = new ArrayList<>();
+        for (Enrollment enrollment : enrollments) {
+            // not sure what this part covered but it threw an error and works without it
+            // Enrollment grade = null;
+            // if (enrollment.getGrade() !=null) {
+            //     grade = enrollmentRepository.(enrollment.getGrade());
+            // }
```
Comment (bustyAI) [inline src/main/java/com/cst438/controller/StudentController.java:56]: Could delete if no longer needed

File: src/main/java/com/cst438/controller/StudentController.java [hunk 3]
```diff
@@ -16,66 +20,182 @@
...
+            @RequestParam("studentId") int studentId) {
 
         // TODO
-
         // check that the Section entity with primary key sectionNo exists
         // check that today is between addDate and addDeadline for the section
         // check that student is not already enrolled into this section
-        // create a new enrollment entity and save.  The enrollment grade will
+        // create a new enrollment entity and save. The enrollment grade will
         // be NULL until instructor enters final grades for the course.
```
Comment (bustyAI) [inline src/main/java/com/cst438/controller/StudentController.java:128]: Could delete if no longer needed

File: src/main/java/com/cst438/controller/StudentController.java [hunk 4]
```diff
@@ -16,66 +20,182 @@
...
-   @DeleteMapping("/enrollments/{enrollmentId}")
-   public void dropCourse(@PathVariable("enrollmentId") int enrollmentId) {
-
-       // TODO
-       // check that today is not after the dropDeadline for section
-   }
+    @DeleteMapping("/enrollments/{enrollmentId}")
+    public void dropCourse(@PathVariable("enrollmentId") int enrollmentId) {
+        // TODO
+        // check that today is not after the dropDeadline for section
```
Comment (bustyAI) [inline src/main/java/com/cst438/controller/StudentController.java:182]: Could delete if no longer needed

File: src/main/java/com/cst438/domain/CourseRepository.java [hunk 5]
```diff
@@ -1,9 +1,15 @@
...
 import java.util.List;
 
 public interface CourseRepository extends CrudRepository<Course, String> {
 	
     List<Course> findAllByOrderByCourseIdAsc();
+
+    // was shown in the lecture but isn't up to date with the current database format
+    // @Query("select distinct s.course from Term t join t.sections s " +
+    //         "where t.year=:year and t.semester=:semester order by s.course.courseId")
+    //List<Course> findByYearAndSemesterOrderByCourseId(int year, String semester);
```
Comment (bustyAI) [inline src/main/java/com/cst438/domain/CourseRepository.java:14]: Could delete if no longer needed

**Review by bustyAI — COMMENTED** [review]
Would be fine to delete the commented lines of code that are not needed.

**Comment by Nireves333** [comment]
Tested EnrollmentController and AssignmentController with Postman and the H2 Console. No errors were found

**Comment by ChrisVarela** [comment]
Tested on local IDE
- EnrollmentController
- Assignment Controller
no issues found

**Comment by jeremiahmcgrath** [comment]
Hey y'all! Looks good from my end, I found no errors when testing it locally.

**Pull request merged**